### PR TITLE
SONAR-7665 Deprecate @BatchSide and introduce @ScannerSide

### PR DESCRIPTION
**Pull request by dbmeneses**
I didn't change sonar-plugin-api to ensure backward compatibility if plugins use the API in older instances.

### CODE REVIEW

File: sonar-scanner-engine/src/main/java/org/sonar/batch/bootstrap/ExtensionUtils.java [hunk 1]
```diff
@@ -36,9 +37,10 @@ public static boolean isInstantiationStrategy(Object extension, String strategy)
     }
     return InstantiationStrategy.PER_PROJECT.equals(strategy);
   }
-
-  public static boolean isBatchSide(Object extension) {
-    return AnnotationUtils.getAnnotation(extension, BatchSide.class) != null;
+  
+  public static boolean isScannerSide(Object extension) {
+    return AnnotationUtils.getAnnotation(extension, BatchSide.class) != null ||
```
Comment (SonarTech) [inline sonar-scanner-engine/src/main/java/org/sonar/batch/bootstrap/ExtensionUtils.java]: MINOR Remove this use of "BatchSide"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. INFO [BatchSide.java#L52](https://github.com/SonarSource/sonarqube/blob/edcb452ef5232ea65b2eca7b58cfd8831c935216/sonar-plugin-api/src/main/java/org/sonar/api/batch/BatchSide.java#L52): Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

**Comment by henryju** [comment]
@dbmeneses API is provided at runtime so it should not be an issue

**Comment by dbmeneses** [comment]
ok @henryju, i've updated it

File: sonar-plugin-api/src/main/java/org/sonar/api/batch/ScannerSide.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+package org.sonar.api.batch;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Marker annotation for all the components available in container of batch (code analyzer). Note that
```
Comment (henryju) [inline sonar-plugin-api/src/main/java/org/sonar/api/batch/ScannerSide.java]: s/batch/scanner

**Comment by henryju** [comment]
Minor Javadoc glitch then LGTM

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. INFO [BatchSide.java#L52](https://github.com/SonarSource/sonarqube/blob/93643e8a809b13b80aa1029e6198dad43c85e8bd/sonar-plugin-api/src/main/java/org/sonar/api/batch/BatchSide.java#L52): Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

**Comment by dbmeneses** [comment]
Merged

**Pull request closed without merging**